pkg/asset/machines/openstack/machines.go

```diff
@@ -302,6 +302,32 @@ func failureDomainsFromSpec(mpool openstack.MachinePool) []machinev1.OpenStackFa
 			if volumes := len(mpool.RootVolume.Zones); volumes > numberOfFailureDomains {
 				numberOfFailureDomains = volumes
 			}
+
+			// When distributing the Root volumes across multiple failure domains, we must use multiple Storage types so they use a different backend.
+			// Storage availability zones are purely cosmetic for now and can also be used to define where a volume should be created, however
+			// we don't want to force a user to define a Storage availability zone when using multiple Storage types.
+			// Therefore we decided to require as many Storage types as there are Compute availability zones, if there are multiple Compute availability zones.
+			// e.g. it would not make sense to have 3 Compute availability zones and 2 Storage types, because one of the Storage types would be used twice and
+			// therefore the number of failure domains would not be 3 anymore.
+			if mpool.RootVolume.Types != nil {
+				if computes, volumes := len(mpool.Zones), len(mpool.RootVolume.Types); computes > 1 && volumes > 1 && computes != volumes {
```

Comment:
A corresponding check should also exist in the static validations.

Reply:
Good point. I'll address that.